.asf.yaml

```diff
@@ -28,28 +32,19 @@ github:
     squash: true
     merge: false
     rebase: false
-#  protected_branches:
-#    main:
-#      required_status_checks:
-        # strict means "Require branches to be up to date before merging."
-#        strict: true
-
-#      required_pull_request_reviews:
-#        dismiss_stale_reviews: false
-#        required_approving_review_count: 1
-
-      # squash or rebase must be allowed in the repo for this setting to be set to true.
-#      required_linear_history: false
-
-#      required_signatures: false
-
-      # requires all conversations to be resolved before merging is possible
-#      required_conversation_resolution: false
+  del_branch_on_merge: true
+  protected_branches:
+    main:
+      required_status_checks:
+        contexts:
+          - Required
```

Comment:
It should be "the names of checks". But what "the name" is is vague.

I checked my previous work on [Apache Pulsar](https://github.com/apache/pulsar/blob/9937d22ffea22b199fd5782153ebb4a9c2f25391/.asf.yaml#L44-L51). Hopefully, we can get it right the first time.